fw/Core/Hitcon/App/TamaApp.h

```diff
@@ -0,0 +1,206 @@
+#ifndef TAMA_APP_H
+#define TAMA_APP_H
+
+#include <Logic/Display/display.h>
+#include <Service/Sched/PeriodicTask.h>
+#include <Service/Sched/SysTimer.h>  // For SysTimer
+
+#include "app.h"
+
+namespace hitcon {
+namespace app {
+namespace tama {
+
+enum class TAMA_APP_STATE : uint8_t {
+  INTRO_TEXT,   // Displaying introductory text
+  CHOOSE_TYPE,  // Player is selecting a pet type
+  EGG,          // Pet is an egg, waiting to hatch
+  ALIVE,        // Pet has hatched and is alive
+  // TODO: Add states like DEAD, EVOLVING, etc.
+};
+
+enum class TAMA_TYPE : uint8_t {
+  NONE_TYPE,  // No pet selected or game reset
+  DOG,        // Player chose a dog
+  CAT,        // Player chose a cat
+};
+
+typedef struct {
+  TAMA_APP_STATE state;
+  TAMA_TYPE type;
+  unsigned int state_enter_time_ms;  // Timestamp when the current state (e.g.,
+                                     // EGG) was entered
+  // TODO: Add more stats for ALIVE state:
+  // uint32_t birth_time_ms; // Actual hatch time
+  // uint8_t hunger;
+  // uint8_t happiness;
+  // uint32_t last_interaction_time_ms;
+} tama_storage_t;
+
+typedef struct {
+  display_buf_t fb[12][DISPLAY_WIDTH];
+  uint8_t fb_size;
+  uint8_t active_frame;
+} tama_display_fb_t;
+
+class TamaApp : public App {
+ private:
+  static constexpr unsigned ROUTINE_INTERVAL_MS =
+      500;  // How often the Routine function runs
+  TAMA_TYPE _current_selection_in_choose_mode;
+  hitcon::service::sched::PeriodicTask _routine_task;
+  tama_storage_t& _tama_data;
+  tama_display_fb_t _fb;
+
+  void Render();
+  void Routine(void* unused);
+  void UpdateFrameBuffer();
+
+ public:
+  TamaApp();
+  virtual ~TamaApp() = default;
+  void Init();
+
+  void OnEntry() override;
+  void OnExit() override;
+  void OnButton(button_t button) override;
+  void OnEdgeButton(button_t button) override;
+};
+
+extern TamaApp tama_app;
+
+// --- Animation Frame Data Definitions (for tama_ani_t and static asserts) ---
+// Each frame is 8 columns wide. Characters are 5 columns wide and centered.
+
+constexpr display_buf_t TAMA_DOG_FRAMES[2][8] = {
+    {
+        /*
+         * code=68, hex=0x44, ascii="D"
```

Comment:
Why DG instead of DOG?

Reply:
It's a placeholder until we have normal animation XD